wordpress-mobile/WordPress-iOS
# WebViews Auth issue: Recover from disabled auth token

**Fixes** https://github.com/wordpress-mobile/WordPress-iOS/issues/15611

## Description:
The error logged in #15611 was the result of `r-login` being loaded with a stale auth cookie which the app provides to prevent having to fetch a fresh cookie each time. However, there are actions that can be taken that invalidate those cookies on the server but the app mechanisms to clean them up weren't in place. This changeset allows the request to go then tries again if it runs into some errors. 

## To test:
### Setup
You'll want a simple site with a custom-mapped domain name and the ability to invalidate the web token in the web view.

You can invalidate the token either by enabling 2FA or using some of the internal tools.

### Test
1. Log in to the app and open a Web view to make sure your Auth tokens are populated.
2. Invalidate your token.
3. Navigate to My Site > View Site
    - You can also check by Navigating to My Site > Posts > View (on one of the posts) 
4. Expect your Web View to load.

## Regression Notes
1. Potential unintended areas of impact
The app has limited indications that could signal that the cookie is bad so we could be cleaning it up too regularly or for issues not related to authentication. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I checked the other web views from this scenario and wasn't able to produce any oddities.

3. What automated tests I added (or what prevented me from doing so)
Added tests around the flows for when we want to clear the cookies.

## PR submission checklist:
